components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/TypeEncoders.scala

```diff
@@ -145,7 +148,14 @@ class TypingResultDecoder(loadClass: String => Class[_]) {
   }
 
   private def typedUnion(obj: HCursor): Decoder.Result[TypingResult] = {
-    obj.downField("union").as[Set[SingleTypingResult]].map(TypedUnion)
+    obj.downField("union").as[List[SingleTypingResult]].flatMap { list =>
+      NonEmptyList
+        .fromList(list)
+        .map(nel => Right(Typed(nel)))
+        .getOrElse(
+          Left(DecodingFailure(s"Union should has at least 2 elements but it has ${list.size} elements", obj.history))
+        )
+    }
```

Comment:
It's a little trick here. I accept situation when someone pass union with one type during decoding and than I don't produce union but e.g. `TypedClass`. Even if I accept only >= 2 elements I still would rather use our smart "folding" `Typed.apply` because I want to guaranty that in `TypedUnion` there are only types that can't be folded. So we have two options:
- be restrictive and do Typed.apply(...) and check if the result is an union which types the same as in input
- be loose and accept that the result not always be the union

I think that being loose is more safe and rather shouldn't introduce errors. WDYT?

Reply:
Oh, I see. You are right - from those two options loose is the better one - when somebody passes union with one type it is most likely unintended mistake, refactor typo or just unawareness. From NU point of view it is a good idea to treat it as single type. :+1:

Reply:
I added a clarification comment